bytecomp/lambda.mli

```diff
@@ -158,6 +155,9 @@ and array_kind =
 and value_kind =
     Pgenval | Pfloatval | Pboxedintval of boxed_integer
 
+and block_shape =
+  value_kind list option
```

Comment:
What about using array instead of list here?  This would be a little bit more compact, and it would be more appropriate (also lookup-wise) if those block_shape are computed for blocks representing module structures (which can be rather long).